src/services/sync/strategies/TSP/TSPSchoolSyncer.js

```diff
@@ -352,21 +354,41 @@ class TSPSchoolSyncer extends mix(Syncer).with(ClassImporter) {
 	 */
 	async updateTeacher(user, tspTeacher) {
 		try {
+			const userUpdateObject = {};
+
+			// Check if anything has changed on the TSP side regarding user data
+			// and, if yes, set all the current data in the user update object.
 			const equal =
 				(user.namePrefix === tspTeacher.lehrerTitel || (!user.namePrefix && !tspTeacher.namePrefix)) &&
 				user.firstName === tspTeacher.lehrerVorname &&
 				user.lastName === tspTeacher.lehrerNachname;
-			if (equal) {
+			if (!equal) {
+				userUpdateObject.namePrefix = tspTeacher.lehrerTitel;
+				userUpdateObject.firstName = tspTeacher.lehrerVorname;
+				userUpdateObject.lastName = tspTeacher.lehrerNachname;
+			}
+
+			// If the feature flag is enabled, add the last synced at field
+			// to the user options (for the updated user document) to set
+			// the last sync date value to the current date.
+			if (lastSyncedAtEnabled) {
+				userUpdateObject.lastSyncedAt = new Date();
+			}
+
+			// Check if the user update object is empty which would mean that both:
+			//   1. no data has been changed on the TSP side;
+			//   2. "last synced at" feature flag is disabled.
+			// If any of these two is not true, the user object will be updated.
+			if (_.isEmpty(userUpdateObject)) {
```

Comment:
AC: 
2. Information on last sync is stored for every user (even when nothing else changed)

we want to know when the sync has run for the user, regadless if the user is changed or not
this creates indeed performance concern, but should be tried out via the feature flag.